components/common/header/index.tsx

```diff
@@ -1,14 +1,109 @@
-import { client } from '@/sanity/lib/client';
-import { FETCH_HEADER } from '@/sanity/queries/header/fetch-header';
-import { FETCH_HEADERResult } from '@/types/generated/sanity.types';
-
-export interface IHeaderProps {}
-
-export default async function Header(_props: IHeaderProps) {
-  const headerData = await client.fetch<FETCH_HEADERResult>(FETCH_HEADER);
-  // FIXME: Remove this log after done with the example
-  // eslint-disable-next-line no-console
-  console.log('headerData: ', headerData);
-
-  return <header></header>;
-}
+'use client';
+import imageUrlBuilder from '@sanity/image-url';
+import Link from 'next/link';
+import React, { useEffect, useState } from 'react';
+
+import { client } from '@/sanity/lib/client';
+import { FETCH_HEADER } from '@/sanity/queries/header/fetch-header';
+import { FETCH_HEADERResult } from '@/types/generated/sanity.types';
+
+export interface IHeaderProps {}
+
+export default function Header(_props: IHeaderProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  // eslint-disable-next-line no-unused-vars
+  const [headerData, setHeaderData] = useState<FETCH_HEADERResult>(null);
+  const [logoUrl, setLogoUrl] = useState('');
+
+  useEffect(() => {
+    const fetchHeaderData = async () => {
+      const headerDataResult = await client.fetch<FETCH_HEADERResult>(FETCH_HEADER);
+      setHeaderData(headerDataResult);
+      // FIXME: Remove this log after done with the example
+      // eslint-disable-next-line no-console
+      // console.log('headerDataResult: ', headerDataResult);
+      const logoAsset = headerDataResult?.logo?.asset;
+      if (logoAsset !== undefined) {
+        const imageUrl = imageUrlBuilder(client).image(logoAsset._ref).url();
+        setLogoUrl(imageUrl);
+      } else {
+        // eslint-disable-next-line no-console
+        console.error('Logo asset is undefined');
+      }

```

Comment:
Extract this into a util function.
This function should get the asset and return a URL. It should handle errors as well.
Add JSDoc comments to describe what the function does.